### DISCUSSION TITLE
Function to generate the truncated deployment URL

### DISCUSSION THREAD

**Discussion by kewers** [vercel/community#1950] [Help] [answered]
### Summary

[In the docs about truncated Generated URLs](https://vercel.com/docs/concepts/deployments/generated-urls#truncation) you mention the pattern that should be used to create the truncated URL. 

Does vercel have a function or command in the CLI that does this job for us?

**Comment by mmachatschek** [comment]
We would be interested in this as well. Exposing a helper function to generate the deployment url with builds would be super helpful for interconnected apps

**Comment by swarnava** [comment, marked as answer]
Hey,

This truncated URL should be generated automatically. You can refer to docs example to determine how to detect the automated URL from a deployment.

**Reply by mmachatschek** [reply]
@swarnava unfortunately this doesn't solve our usecase. also to my understanding, this doesn't answer @kewers question

**Comment by k9ordon** [comment]
We have encountered a similar issue. 

We would like to link different preview deployments of a monorepo together. Specifically, we have an API backend that we want to reference in our frontend project. Therefore, we need the deployment URL of the API project in the frontend project environment. Our plan is to reconstruct the branch URL of the API project during the start command of the frontend to specify the API endpoint. 

From the documentation and the discussion at https://github.com/orgs/vercel/discussions/472#discussioncomment-5294579, it is possible to reverse engineer the creation of the branch URL. However, a more future-proof solution would be a package/helper exposed by Vercel.
